internal/api/impl/v1/view/endpoint.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package view
+
+import (
+	"fmt"
+
+	"github.com/labstack/echo/v4"
+	apiInterface "github.com/perses/perses/internal/api/interface"
+	"github.com/perses/perses/internal/api/interface/v1/view"
+	"github.com/perses/perses/internal/api/rbac"
+	"github.com/perses/perses/internal/api/route"
+	"github.com/perses/perses/internal/api/toolbox"
+	"github.com/perses/perses/internal/api/utils"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/perses/perses/pkg/model/api/v1/role"
+)
+
+type endpoint struct {
+	service view.Service
+	toolbox toolbox.Toolbox
+}
+
+func NewEndpoint(service view.Service, rbac rbac.RBAC, caseSensitive bool) route.Endpoint {
+	return &endpoint{
+		service: service,
+		toolbox: toolbox.New(nil, rbac, v1.KindDashboard, caseSensitive),
+	}
+}
+
+func (e *endpoint) CollectRoutes(g *route.Group) {
+	g.POST(fmt.Sprintf("/%s", utils.PathView), e.View, false)
+}
+
+func (e *endpoint) View(ctx echo.Context) error {
+	view := v1.View{}
+	if err := ctx.Bind(&view); err != nil {
+		return apiInterface.HandleBadRequestError(err.Error())
+	}
+
+	if err := view.Validate(); err != nil {
+		return apiInterface.HandleBadRequestError(err.Error())
+	}
+
+	if err := e.toolbox.CheckPermission(ctx, nil, apiInterface.Parameters{
```

Comment:
As @Gladorme  suggested me, it's simpler to use directly RBAC service actually 

Here is how the code would looks like 
```go
  claims := crypto.ExtractJWTClaims(ctx)
	if claims == nil {
		// Claims can be nil with anonymous endpoint and unauthenticated users, no need to continue
		return nil
	}
	if ok := e.rbac.HasPermission(claims.Subject, role.ReadAction, view.Project, role.DashboardScope); !ok {
		return apiInterface.HandleUnauthorizedError(fmt.Sprintf("missing '%s' permission in '%s' project for '%s' kind", role.ReadAction, view.Project, role.DashboardScope))
	}
```

Reply:
The CheckPermissions from toolbox has a lot of complicate conditions that doesn't apply with your need

Reply:
Nice. That's a lot cleaner - thanks!